src/dataframerow/dataframerow.jl

```diff
@@ -62,11 +62,13 @@ Base.values(r::DataFrameRow) = ntuple(col -> parent(r)[col][row(r)], length(r))
 
 # hash column element
 Base.@propagate_inbounds hash_colel(v::AbstractArray, i, h::UInt = zero(UInt)) = hash(v[i], h)
-Base.@propagate_inbounds hash_colel(v::AbstractCategoricalArray, i, h::UInt = zero(UInt)) =
-    hash(CategoricalArrays.index(v.pool)[v.refs[i]], h)
 Base.@propagate_inbounds function hash_colel(v::AbstractCategoricalArray{>: Missing}, i, h::UInt = zero(UInt))
     ref = v.refs[i]
-    ref == 0 ? hash(missing, h) : hash(CategoricalArrays.index(v.pool)[ref], h)
+    if eltype(v) >: Missing && ref == 0
+        hash(missing, h)
+    else
+        hash(CategoricalArrays.index(v.pool)[ref], h)
```

Comment:
I understand that here `ref` will never be `0` right?

Reply:
Yes (at least if the array isn't corrupt). This is really equivalent to the previous code, except that the type check is inside the function now.